index.js

```diff
@@ -1251,17 +1285,59 @@ module.exports = function (log, indexesPath) {
     }
   }
 
+  // Scanning the log for this purpose is much slower than the keys index in
+  // ssb-db2, but we don't need this to be fast because it's a one-time job only
+  // after compaction has ended. We should use ssb-db2 `keys` in the future.
+  function findSeqForMsgKey(msgKey, cb) {
+    let seq = 0
+    log.stream({ offsets: false, values: true, decrypt: false }).pipe(
+      push.drain(
+        function sinkToFindSeq(buffer) {
+          const pKey = bipf.seekKey2(buffer, 0, BIPF_KEY, 0)
+          const msgKeyCandidate = bipf.decode(buffer, pKey)
+          if (msgKeyCandidate === msgKey) {
+            cb(null, seq)
+            return false // abort sink
+          } else {
+            seq += 1
+          }
+        },
+        function sinkEndedLookingForSeq() {
+          cb(new Error('msgKey not found'))
+        }
+      )
+    )
+  }
+
   function countBitsetSlice(bitset, seq, descending) {
     if (!seq) return bitset.size()
     else return bitset.size() - seq
   }
 
-  function paginate(operation, seq, limit, descending, onlyOffset, sortBy, cb) {
+  function paginate(
+    operation,
+    seq,
+    limit,
+    descending,
+    onlyOffset,
+    sortBy,
+    latestMsgKey,
```

Comment:
Can we call this compactionLatestMsgKey or something like that? To signal that this is only ever used for compaction.